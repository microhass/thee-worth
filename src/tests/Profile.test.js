import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Profile from '../components/profile/Profile';

const mockStore = configureMockStore();
describe('Profile Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      missions: {
        missions: [
          {
            id: 1,
            name: 'Mission 1',
            isMember: false,
          },
          {
            id: 2,
            name: 'Mission 2',
            isMember: true,
          },
        ],
      },
      rockets: {
        rockets: [
          {
            id: 1,
            name: 'Rocket 1',
            reserved: true,
          },
          {
            id: 2,
            name: 'Rocket 2',
            reserved: true,
          },
        ],
      },
    });
  });

  it('should render the profile page', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
