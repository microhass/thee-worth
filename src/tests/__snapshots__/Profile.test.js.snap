// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component should render the profile page 1`] = `
<DocumentFragment>
  <section
    class="profile"
  >
    <div
      class="profile-missions"
    >
      <h3
        class="title"
      >
        my missions
      </h3>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <span>
            Mission 2
          </span>
          <button
            class="leave"
            type="button"
          >
            leave mission
          </button>
        </li>
      </ul>
    </div>
    <div
      class="profile-rockets"
    >
      <h3
        class="title"
      >
        my rockets
      </h3>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <span>
            Rocket 1
          </span>
          <button
            class="leave"
            type="button"
          >
            cancel reservation
          </button>
        </li>
        <li
          class="item"
        >
          <span>
            Rocket 2
          </span>
          <button
            class="leave"
            type="button"
          >
            cancel reservation
          </button>
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
